Source/WebCore/rendering/svg/SVGRenderingContext.cpp

```diff
@@ -160,24 +160,27 @@ void SVGRenderingContext::prepareToRenderSVGContent(RenderElement& renderer, Pai
     }
 
     if (!isRenderingMask) {
```

Comment:
Maybe it is easier to `&&` the new clause `!renderer.isSVGRootOrLegacySVGRoot()` here instead of indenting all the code below:

```
// RenderLayer should take care of CSS filters applied to SVG roots, so don't reapply it here.
if (!isRenderingMask && !renderer.isSVGRootOrLegacySVGRoot()) {
    ...
}
```

Reply:
I think it is also important to mention in the comment that any filter which is applied to the SVG root element is applied as a CSSFilter even it just references a `<filter>` element in the same SVG.